local/owr_local_media_source.c

```diff
@@ -143,20 +168,35 @@ static void owr_message_origin_interface_init(OwrMessageOriginInterface *interfa
 static void owr_local_media_source_init(OwrLocalMediaSource *source)
 {
     OwrLocalMediaSourcePrivate *priv;
+
     source->priv = priv = OWR_LOCAL_MEDIA_SOURCE_GET_PRIVATE(source);
     priv->device_index = -1;
     priv->message_origin_bus_set = owr_message_origin_bus_set_new();
+    priv->source_volume = NULL;
+    priv->tmp_volume = 1;
 }
 
 static void owr_local_media_source_set_property(GObject *object, guint property_id,
     const GValue *value, GParamSpec *pspec)
 {
     OwrLocalMediaSource *source = OWR_LOCAL_MEDIA_SOURCE(object);
+    OwrMediaType media_type = OWR_MEDIA_TYPE_UNKNOWN;
 
     switch (property_id) {
     case PROP_DEVICE_INDEX:
         source->priv->device_index = g_value_get_int(value);
         break;
+    case PROP_VOLUME:
+        g_object_get(source, "media-type", &media_type, NULL);
+        if (source->priv->source_volume)
+            g_object_set(source->priv->source_volume, "volume", g_value_get_double(value), NULL);
+        
+        if (media_type == OWR_MEDIA_TYPE_AUDIO)
```

Comment:
No.. this way tmp_volume continues to be updated so volume will be preserved across shutdowns.